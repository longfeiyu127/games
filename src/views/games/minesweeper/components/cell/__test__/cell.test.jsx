import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Cell from '../cell.tsx'

Enzyme.configure({ adapter: new Adapter() })

const testData = ['🔄', '🚩', '🔍', '⬜', '1️⃣', '2️⃣', '3️⃣', '4️⃣', '5️⃣', '6️⃣', '7️⃣', '8️⃣', '⬛', '🚩', '💥', '💣']

describe('<Cell />', () => {
  it('component minesweeper/Cell render', () => {
    const Dom = renderer
      .create(
        <div>
          {testData.map((item, index) => (
            <Cell type={item} cols={index} rows={index} key={index} />
          ))}
        </div>
      )
      .toJSON()
    expect(Dom).toMatchSnapshot()
  })

  it('component minesweeper/Cell click event', () => {
    const onClick = jest.fn()
    const cell = mount(<Cell type="🔄" onClick={onClick} cols={-1} rows={-1} key={-1} />)
    cell.find('div').simulate('click')
    expect(onClick.mock.calls.length).toBe(1)
    cell.find('div').simulate('click')
    expect(onClick.mock.calls.length).toBe(2)
  })
})
