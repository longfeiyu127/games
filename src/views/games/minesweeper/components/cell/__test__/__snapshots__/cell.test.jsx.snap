// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> component minesweeper/Cell render 1`] = `
<div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    🔄
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    🚩
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    🔍
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    ⬜
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    1️⃣
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    2️⃣
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    3️⃣
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    4️⃣
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    5️⃣
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    6️⃣
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    7️⃣
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    8️⃣
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    ⬛
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    🚩
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    💥
  </div>
  <div
    className="c-Cell"
    onClick={[Function]}
  >
    💣
  </div>
</div>
`;
